### PR TITLE
Add a rake task to create a superadmin

### LINKED ISSUE #1242: Rake task to create global (non-tenant) user
The current method for creating a super-admin user to manage tenants, requires signing up for a global user account first: https://github.com/samvera-labs/hyku/wiki/Create-super-admin-user.

Now that https://github.com/samvera-labs/hyku/issues/1203 is complete, disabling open sign-up on stack build also removes the ability to sign up as a global user. There needs to be another method for creating these global user accounts. The simplest option is likely a rake task, but that may need discussion.

### PR DESCRIPTION
**Pull request by jcoyne**
Fixes #1242

@projecthydra-labs/hyrax-code-reviewers

### CODE REVIEW

File: spec/tasks/rake_spec.rb [hunk 1]
```diff
@@ -12,21 +12,21 @@
     end
 
     it 'requires user_list argument' do
-      expect { run_task('superadmin:grant') }.to raise_error(ArgumentError)
+      expect { run_task('hyrax:superadmin:grant') }.to raise_error(ArgumentError)
```
Comment (mjgiarlo) [inline spec/tasks/rake_spec.rb]: Shouldn't these all be `hyku:`?

File: lib/tasks/user.rake [hunk 2]
```diff
@@ -0,0 +1,38 @@
+namespace :hyku do
+  namespace :superadmin do
+    desc 'Create an initial superadmin user'
```
Comment (mjgiarlo) [inline lib/tasks/user.rake]: `s/an initial/a/`?

File: lib/tasks/user.rake [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    task create: :environment do
+      puts 'Creating an initial superadmin user.'
```
Comment (mjgiarlo) [inline lib/tasks/user.rake]: `s/an initial/a/`?

File: lib/tasks/user.rake [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      user = prompt_to_create_user
+      user.add_role(:superadmin)
+      puts 'User created.'
+    end
+
+    def prompt_to_create_user
```
Comment (mjgiarlo) [inline lib/tasks/user.rake]: So, am I reading this PR right that we now have two rake tasks that can elevate an existing user to superadmin? I'm surprised the new task works for existing users given we already have the above rake task for that. Maybe I'm misreading this.

File: lib/tasks/user.rake [hunk 5]
```diff
@@ -0,0 +1,38 @@
+namespace :hyku do
+  namespace :superadmin do
+    desc 'Create an initial superadmin user'
+    task create: :environment do
+      puts 'Creating an initial superadmin user.'
+      user = prompt_to_create_user
+      user.add_role(:superadmin)
+      puts 'User created.'
```
Comment (mjgiarlo) [inline lib/tasks/user.rake]: How about `puts "User (#{user.user_key}) created and granted superadministrator privileges"`?

File: spec/tasks/rake_spec.rb [hunk 6]
```diff
@@ -12,21 +12,21 @@
     end
```
Comment (mjgiarlo) [inline spec/tasks/rake_spec.rb]: Should test coverage be added here for the new `hyku:superadmin:create` task?

File: lib/tasks/user.rake (hunk 4 above)
Comment (jcoyne) [inline lib/tasks/user.rake]: No, this makes a new user and elevates them. The other elevates an existing user.

Comment (mjgiarlo) [inline lib/tasks/user.rake]: @jcoyne If it only creates new users, do we need `#find_or_create_by` on line 12? The `find` there suggests that this task can act on existing users.

Comment (jcoyne) [inline lib/tasks/user.rake]: Let me rephrase, this elevates a user.  If they don't exist it creates them.  I believe this is preferable to: "Nope, that user exists. Go find a different rake task to do what you want"

Comment (mjgiarlo) [inline lib/tasks/user.rake]: @jcoyne Sounds good. Should we retire the other rake task as part of this branch?

Comment (jcoyne) [inline lib/tasks/user.rake]: @mjgiarlo your call. That one takes a list of users and elevates, this does one at a time but can create them & set passwords too.

File: spec/tasks/rake_spec.rb (hunk 6 above)
Comment (jcoyne) [inline spec/tasks/rake_spec.rb]: Honestly, I don't know how to test stuff that's reading from `$stdin`.  This code is copied from https://github.com/projectblacklight/spotlight/blob/master/lib/tasks/spotlight_tasks.rake#L2 and it wasn't tested their either.

File: lib/tasks/user.rake (hunk 4 above)
Comment (mjgiarlo) [inline lib/tasks/user.rake]: Since you're adding functionality, my preference is to go with this new rake task for both use cases.
